src/python/pants/rules/core/list_target_types.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Type
+
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem, LineOriented
+from pants.engine.rules import UnionMembership, goal_rule
+from pants.engine.target import BoolField, Field, RegisteredTargetTypes, Target
+from pants.util.objects import get_first_line_of_docstring
+
+
+class TargetTypesOptions(LineOriented, GoalSubsystem):
+    """List all the registered target types, including custom plugin types."""
+
+    # TODO: consider renaming this to to `target-types` or even a more general goal like
```

Comment:
I like `target-types` because it clarifies that this isn't `list`...

Reply:
Me too. My vote is for target-types. Stu proposed `symbols` in https://pantsbuild.slack.com/archives/C0D7TNJHL/p1584985449127300 so I punted on this for now. 

Before we require writing bindings, we’ll need to decide the name for this and things like the `targets()` registration entry point. So, we’ll decide this soon.

Reply:
I had to go with `target-types` because `targets2` gets renamed to `targets` when in V2-only mode and that caused a duplicate option scope.

I used `target-types2` to reflect that this could still change. We can fix the name at the same time we decide what to do with `targets2`.